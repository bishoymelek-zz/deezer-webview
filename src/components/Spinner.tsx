import React from 'react';
import { Spin } from 'antd';
import styled from 'styled-components';

type SpinnerPropType = { spinning: boolean; delay?: number; className: string };

function Spinner(props: SpinnerPropType): JSX.Element {
  const { spinning, delay = 1000, className } = props;
  if (spinning) {
    return (
      <div className={`spinner-wrapper ${className}`}>
        <Spin spinning size="large" wrapperClassName="comp" delay={delay} />
      </div>
    );
  }
  return <></>;
}
const StyledSpinner = styled(Spinner)`
  &.spinner-wrapper {
    width: 100%;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    position: fixed;
    background-color: rgba(255, 255, 255, 0.7);
    z-index: 99999;

    .ant-spin {
      z-index: 9991;
      margin: auto;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }
`;

export default StyledSpinner;
export { StyledSpinner };
