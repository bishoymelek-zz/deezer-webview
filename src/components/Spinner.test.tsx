import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Spinner from './Spinner';

const mockStore = configureStore([]);

describe('Component/Spinner', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      myState: 'sample text',
    });

    component = renderer.create(
      <Provider store={store}>
        <Spinner />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
