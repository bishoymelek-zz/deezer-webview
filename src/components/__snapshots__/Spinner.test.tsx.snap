// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/Spinner should render with given state from Redux store 1`] = `null`;
